l2geth/core/types/receipt_test.go

```diff
@@ -281,3 +281,145 @@ func clearComputedFieldsOnLog(t *testing.T, log *Log) {
+		TransactionIndex: math.MaxUint32,
+		L1GasPrice:       big.NewInt(0),
+		L1GasUsed:        big.NewInt(int64(100)),
+		L1Fee:            big.NewInt(int64(100)),
+		FeeScalar:        FeeScalar,
+		DAGasUsed:        big.NewInt(int64(10)),
+		DAFee:            &big.Int{},
+		DAGasPrice:       big.NewInt(int64(10)),
+	}
+	receipt.Bloom = CreateBloom(Receipts{receipt})
+
+	receiptByte, err := encodeAsStoredReceiptRLPV2(receipt)
+	if err != nil {
+		t.Fatalf("Error encoding receipt: %v", err)
+		return
+	}
+	var s ReceiptForStorage
+	err = rlp.DecodeReceiptsBytes(receiptByte, &s)
+	if err != nil {
+		t.Fatalf("Error decoding RLP receipt: %v", err)
+	}
+
+	t.Log(receipt)
+	t.Log(s)
+
+	if !bytes.Equal(receipt.PostState, s.PostState) {
+		t.Errorf("receipt.PostState = %v, want %v", receipt.PostState, s.PostState)
+	}
+
+	for i := 0; i < len(s.Logs); i++ {
+		if s.Logs[i].Address != receipt.Logs[i].Address {
+			t.Fatalf("receipt log %d address mismatch, want %v, have %v", i, receipt.Logs[i].Address, s.Logs[i].Address)
+		}
+	}
+	if receipt.CumulativeGasUsed != s.CumulativeGasUsed {
+		t.Errorf("receipt.CumulativeGasUsed = %v, want %v", receipt.CumulativeGasUsed, s.CumulativeGasUsed)
+	}
+	if receipt.Bloom != s.Bloom {
+		t.Errorf("receipt.Bloom = %v, want %v", receipt.Bloom, &s.Bloom)
+	}
+	if receipt.L1GasPrice.String() != s.L1GasPrice.String() {
+		t.Errorf("receipt.L1GasPrice = %v, want %v", receipt.L1GasPrice, s.L1GasPrice)
+	}
+	if receipt.L1Fee.String() != s.L1Fee.String() {
+		t.Errorf("receipt.L1Fee = %v, want %v", receipt.L1Fee, s.L1Fee)
+	}
+	if receipt.L1GasUsed.String() != s.L1GasUsed.String() {
+		t.Errorf("receipt.L1GasUsed = %v, want %v", receipt.L1GasUsed, s.L1GasUsed)
+	}
+	if receipt.FeeScalar.String() != s.FeeScalar.String() {
+		t.Errorf("receipt.FeeScalar = %v, want %v", receipt.FeeScalar, s.FeeScalar)
+	}
+	if receipt.DAFee.String() != s.DAFee.String() {
+		t.Errorf("receipt.DAFee = %v, want %v", receipt.DAFee, s.DAFee)
+	}
+	if receipt.DAGasUsed.String() != s.DAGasUsed.String() {
+		t.Errorf("receipt.DAGasUsed = %v, want %v", receipt.DAGasUsed, s.DAGasUsed)
+	}
+	if receipt.DAGasPrice.String() != s.DAGasPrice.String() {
+		t.Errorf("receipt.L1GasPrice = %v, want %v", receipt.DAGasPrice, s.DAGasPrice)
+	}
+
+}
+
+func encodeAsStoredReceiptRLPV2(want *Receipt) ([]byte, error) {
+	stored := storedReceiptRLP{
+		PostStateOrStatus: want.statusEncoding(),
+		CumulativeGasUsed: want.CumulativeGasUsed,
+		Logs:              make([]*LogForStorage, len(want.Logs)),
+		L1GasUsed:         want.L1GasUsed,
+		L1GasPrice:        want.L1GasPrice,
+		L1Fee:             want.L1Fee,
+		FeeScalar:         want.FeeScalar.String(),
+		DAGasUsed:         want.DAGasUsed,
+		DAGasPrice:        want.DAGasPrice,
+		DAFee:             want.DAFee,
+	}
+	for i, log := range want.Logs {
+		stored.Logs[i] = (*LogForStorage)(log)
+	}
+	return rlp.EncodeToBytes(stored)
+}
+
+// testEncodeRLP
+var receiptTests = []Receipt{
```

Comment:
It looks like that `receiptTests` isn't referenced in this file,  what is it used for?
we could add test to coverage the case listed in the receiptTest.

Reply:
ok，I will add tests to cover